src/shogun/base/SGObject.h

```diff
@@ -1011,5 +1021,64 @@ class CSGObject
 	/** Subscriber used to call onNext, onComplete etc.*/
 	SGSubscriber* m_subscriber_params;
 };
+
+#ifndef SWIG
+template <typename T1, typename T2>
+bool execute_dispatch_array_type(const CSGObject* obj, const std::string& name, T2&& lambda)
+{
+	Tag<CDynamicObjectArray*> tag_array_sg(name);
+	if (obj->has(tag_array_sg))
+	{
+		auto dispatched = obj->get(tag_array_sg);
+		lambda(*dispatched); // is stored as a pointer
+		return true;
+	}
+
+	Tag<std::vector<T1*>> tag_vector(name);
+	if (obj->has(tag_vector))
+	{
+		auto dispatched = obj->get(tag_vector);
+		lambda(dispatched);
+		return true;
+	}
+
+	return false;
+}
+
+struct ByName
+{
+};
+
+struct ByNameIndex
+{
+	ByNameIndex(index_t index) : m_index(index) {}
+	index_t m_index;
+};
+
+template <typename T>
+CSGObject* get(const CSGObject* obj, const std::string& name, ByName)
+{
+	return obj->get<T*>(name);
+}
+
+template <typename T>
+CSGObject* get(const CSGObject* obj, const std::string& name, ByNameIndex i)
```

Comment:
detail namespace or less general name

Reply:
I guess `GetByName`, `GetByNameAndIndex`.

This can't be detail namespace as it is directly exposed.

Reply:
i mean it can be still called detail::get? :)

Reply:
Yes was thinking that as well. Next step

Reply:
I have put it all in `sgobject_get_details` now